### PR TITLE
Fixed LIS checks in fast moving vehicles behaving unexpectedly

### PR DESCRIPTION
**Pull request by derbismarck**
Closes https://github.com/diwako/diwako_dui/issues/194

### CODE REVIEW

File: addons/nametags/functions/fnc_onDraw.sqf [hunk 1]
```diff
@@ -16,9 +16,15 @@ if (GVAR(useLIS)) then {
     // unit, per the comment from Pierre MGI at https://community.bistudio.com/wiki/cursorObject
     // we'd like to find the cursorObject regardless, so we piggyback on the LIS check to find the most probable one
 
+    // This is the best we can do without a positionCameraToWorldVisual command.
+    // positionCameraToWorld translates position from camera space to world space in SIMULATION time scope,
+    // while a theoretical positionCameraToWorldVisual command would translate positions from camera space to world
+    // space in RENDER time scope.
+    private _camPos = _player modelToWorldVisualWorld (_player selectionPosition "pilot");
+    private _endPos = _camPos vectorAdd ((getCameraViewDirection _player) vectorMultiply (GVAR(renderDistance) + 1));
```
Comment (diwako) [inline addons/nametags/functions/fnc_onDraw.sqf:24]: Imo there is no need to bind them to a variable. Just immediately use them in the lineIntersectsSurfaces. That way the runtime is a little faster.

**Review by diwako — CHANGES_REQUESTED** [review]
Just tested it. The flickering in fast moving vehicles is still happening. However it is less as it looks like the detection is more accurate to the visual simulation.

Still there is one change request, detailed above.

**Comment by derbismarck** [comment]
Acknowledged -- I was also seeing that flickering but thought it was specific to other changed we'd made, it might have to do with the distance check being performed, I'll take a look in the near future and make the requested change.

**Comment by derbismarck** [comment]
Confirmed that in `diwako_dui_nametags_fnc_calculateFadeValue`, the value of `_distance` is _not_ at fault, presumably because despite the `distance` command not being in visual time scope, the actual offset in position for both the player and the vehicle being checked against is identical. Observe output of the following to see that the error between an exact visual distance check and a `object1 distance object2` check is less than a centimeter, assuming that both `object1` and `object2` are in the same vehicle, or if one is the parent vehicle of the other.

`onEachFrame {systemchat str abs (vectorMagnitude ((getPosASLVisual object1) vectorDiff (getPosASLVisual object2)) - (object1 distance object2))};`

For the case of a player looking at a crewmember of their vehicle, `object1` would usually be `player` and `object2` would be `vehicle player`, at least as it's currently written (more on that down below).

So that wasn't the issue. However, the issue _is_ in `diwako_dui_nametags_fnc_calculateFadeValue` -- it's with the visibility check. I've committed a fix for that that now utilizes visual commands to get eyepos, similarly to the change made in the LIS code. That fixes the flickering for me. I also made a small performance change by removing the definition of `_pos` and instead putting it directly in the `checkVisibility` call. Here's it fixed: https://giant.gfycat.com/BleakActualKagu.mp4


All of that being said, there are still some issues with how `diwako_dui_nametags_fnc_calculateFadeValue` works, which I've already looked at fixing. I'll likely create a separate PR/issue for those. They're mostly to do with the fact that `_distance` is being taken as the distance between `_player` and `vehicle _target` even when `_player` and `_target` are both crew in the same vehicle, leading to unexpectedly high fade values if both the `_player` and the `_target` are in a large vehicle, far from the center of the vehicle, but very close to one another.

**Review by diwako — APPROVED** [review]
Finally gotten around and tested this. It does work as detailed in the post above. Also did some smallish performance testing and it was negligible as well

**Pull request merged**